### PR TITLE
API Functions

### LINKED ISSUE #96: Implement API Functions
**Describe the task that needs to be done.**
Need to change the existing API stubs to call API endpoints from the backend.

**Describe how a solution to your proposed task might look like (and any alternatives considered).**
Will change the API stubs in communication, pairing, timetable and messaging to make calls to the API endpoints in the backend. Also create a utils file to handle repeated HTTP requests

**Notes**
Converting API stubs from #74

### PR DESCRIPTION
**Pull request by sv-singh**
## Description
Converted the previous API stubs to actual API calls based off the backend. Also created helper functions to handle HTTP requests and interceptors.

Fixes #96

@scorpionknifes 
Updated endpoints to match the refractor endpoints here https://github.com/SE701-T1/backend/pull/103
Fixed up socketIO connections with backend using context
Update `.env.example`
Added google-oauth redirect endpoint

## How has this been tested?
Haven't created tests yet

## Screenshots
N/A

## Checklist
*(Leave blank if not applicable)*

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by scorpionknifes** [comment]
Updated endpoints to match the refractor endpoints here https://github.com/SE701-T1/backend/pull/103

**Review by scorpionknifes — APPROVED** [review]
LGTM

File: src/pages/Routes/GoogleRedirectRoute.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      // eslint-disable-next-line no-console
+      console.error(error);
+      navigate('/login', { replace: true });
+    }
+  }, []);
+
+  return <LoadingComponent />;
+}
+
+export default GoogleRedirectRoute;
```
Comment (scorpionknifes) [inline src/pages/Routes/GoogleRedirectRoute.jsx:40]: Added for Google OAuth2 redirect.

set google oauth redirect to http://localhost:3000/google-oauth
e.g http://localhost:3000/google-oauth#access_token=.............................

auto parse access token and send to backend and get new jwt.

**Review by KirstyGong — COMMENTED** [review]
LGTM

**Pull request merged**